advanced_functionality/autogluon-tabular/AutoGluon_Tabular_SageMaker.ipynb

```diff
@@ -0,0 +1,607 @@
+    "                                max_concurrent_transforms=1,                              \n",
+    "                                output_path=output_path)\n",
+    "\n",
+    "transformer.transform(test_s3_path, content_type='text/csv')\n",
+    "transformer.wait()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "outputs": [],
+   "source": [
+    "# Check s3 for <filename>.csv.out file\n",
+    "if instance_type != 'local':\n",
+    "    !aws s3 ls {transformer.output_path} --recursive\n",
+    "elif 's3' in output_path:\n",
+    "    !aws s3 ls {output_path+transformer.latest_transform_job.job_name} --recursive"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "source": [
+    "### Endpoint"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "source": [
+    "##### Deploy remote or local endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "outputs": [],
+   "source": [
+    "instance_type = 'ml.m5.2xlarge'\n",
+    "#instance_type = 'local'\n",
+    "\n",
+    "predictor = model.deploy(initial_instance_count=1, \n",
+    "                         instance_type=instance_type)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "source": [
+    "##### Attach to endpoint (or reattach if kernel was restarted)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "outputs": [],
+   "source": [
+    "# Select standard or local session based on instance_type\n",
+    "if instance_type == 'local': sess = local_session\n",
+    "else: sess = session\n",
+    "\n",
+    "# Attach to endpoint\n",
+    "predictor = AutoGluonTabularPredictor(predictor.endpoint, sagemaker_session=sess)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "source": [
+    "##### Predict on unlabeled test data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "outputs": [],
+   "source": [
+    "results = predictor.predict(X_test.to_csv())\n",
+    "\n",
+    "# Check output\n",
+    "print(Counter(results.splitlines()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "source": [
+    "##### Predict on data that includes label column  \n",
+    "Prediction performance metrics will be printed to endpoint logs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "Collapsed": "false"
+   },
+   "outputs": [],
+   "source": [
+    "results = predictor.predict(test.to_csv())\n",
+    "\n",
+    "# Check output\n",
+    "sleep(0.1); print(Counter(results.splitlines()))"
```

Comment:
Because when running in local mode it ends up outputting the print line before the inference server has finished outputting logs. This just guarantees the print will come last.